### ISSUE TITLE
problem installing python package

### ISSUE THREAD

**Issue by meetrajsinh312** [ggerganov/ggwave#42]
fatal error C1083: Cannot open source file: 'ggwave.bycython.cpp': No such file or directory
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30037\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2

i am recieving this error when installing ggwave in python using "pip install ggwave" . researched a lot but no solution found. maybe one possible solution is install using wheel file but wheel file is not present for ggwave.

same issue with pyaudio but i managed to install it using pipwin install pyaudio.
Another solution is to add ggwave in pipwin package list.

Full error ::::
Collecting ggwave
  Using cached ggwave-0.2.2.tar.gz (62 kB)
Building wheels for collected packages: ggwave
  Building wheel for ggwave (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\users\meetr\appdata\local\programs\python\python38\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-install-azsi740u\\ggwave_5580c82224ed49db82087eca3795e92a\\setup.py'"'"'; __file__='"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-install-azsi740u\\ggwave_5580c82224ed49db82087eca3795e92a\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\Meetr\AppData\Local\Temp\pip-wheel-v9m_y4jz'
       cwd: C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\
  Complete output (73 lines):
  running bdist_wheel
  running build
  running build_ext
  building 'ggwave' extension
  creating build
  creating build\temp.win-amd64-3.8
  creating build\temp.win-amd64-3.8\Release
  creating build\temp.win-amd64-3.8\Release\ggwave
  creating build\temp.win-amd64-3.8\Release\ggwave\src
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Iggwave/include -Iggwave/include/ggwave -Ic:\users\meetr\appdata\local\programs\python\python38\include -Ic:\users\meetr\appdata\local\programs\python\python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt" /EHsc /Tpggwave.bycython.cpp /Fobuild\temp.win-amd64-3.8\Release\ggwave.bycython.obj -O3 -std=c++11
  cl : Command line warning D9002 : ignoring unknown option '-O3'
  cl : Command line warning D9002 : ignoring unknown option '-std=c++11'
  ggwave.bycython.cpp
  ggwave.bycython.cpp(1881): warning C4244: 'argument': conversion from 'Py_ssize_t' to 'int', possible loss of data
  ggwave.bycython.cpp(1923): warning C4244: 'argument': conversion from 'Py_ssize_t' to 'int', possible loss of data
  ggwave.bycython.cpp(2162): warning C4244: 'argument': conversion from 'Py_ssize_t' to 'int', possible loss of data
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Iggwave/include -Iggwave/include/ggwave -Ic:\users\meetr\appdata\local\programs\python\python38\include -Ic:\users\meetr\appdata\local\programs\python\python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt" /EHsc /Tpggwave/src/ggwave.cpp /Fobuild\temp.win-amd64-3.8\Release\ggwave/src/ggwave.obj -O3 -std=c++11
  cl : Command line warning D9002 : ignoring unknown option '-O3'
  cl : Command line warning D9002 : ignoring unknown option '-std=c++11'
  ggwave.cpp
  C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon\gf.hpp(150): warning C4244: 'argument': conversion from 'uint16_t' to 'uint8_t', possible loss of data
  C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon\gf.hpp(150): warning C4244: 'argument': conversion from 'uint16_t' to 'uint8_t', possible loss of data
  C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon\gf.hpp(216): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data
  C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon/rs.hpp(174): warning C4267: 'argument': conversion from 'size_t' to 'uint8_t', possible loss of data
  C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon/rs.hpp(455): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data
  C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon/rs.hpp(500): warning C4267: 'argument': conversion from 'size_t' to 'uint8_t', possible loss of data
  C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon/rs.hpp(519): warning C4267: 'argument': conversion from 'size_t' to 'uint8_t', possible loss of data
  ggwave/src/ggwave.cpp(172): error C2065: 'M_PI': undeclared identifier
  ggwave/src/ggwave.cpp(173): error C2065: 'M_PI': undeclared identifier
  ggwave/src/ggwave.cpp(177): error C2065: 'M_PI': undeclared identifier
  ggwave/src/ggwave.cpp(178): error C2065: 'M_PI': undeclared identifier
  ggwave/src/ggwave.cpp(229): warning C4244: 'initializing': conversion from 'float' to 'int', possible loss of data
  ggwave/src/ggwave.cpp(230): warning C4244: 'initializing': conversion from 'float' to 'int', possible loss of data
  ggwave/src/ggwave.cpp(232): warning C4244: 'initializing': conversion from 'int' to 'float', possible loss of data
  ggwave/src/ggwave.cpp(337): warning C4305: 'initializing': truncation from 'double' to 'float'
  ggwave/src/ggwave.cpp(385): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
  ggwave/src/ggwave.cpp(389): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
  ggwave/src/ggwave.cpp(415): warning C4244: '=': conversion from 'double' to 'float', possible loss of data
  ggwave/src/ggwave.cpp(499): error C2065: 'M_PI': undeclared identifier
  ggwave/src/ggwave.cpp(524): error C2065: 'M_PI': undeclared identifier
  ggwave/src/ggwave.cpp(528): error C2065: 'M_PI': undeclared identifier
  ggwave/src/ggwave.cpp(639): warning C4244: '=': conversion from '_Ty' to '_Ty', possible loss of data
          with
          [
              _Ty=float
          ]
          and
          [
              _Ty=int16_t
          ]
  ggwave/src/ggwave.cpp(649): warning C4244: '=': conversion from '_Ty' to 'uint8_t', possible loss of data
          with
          [
              _Ty=float
          ]
  ggwave/src/ggwave.cpp(656): warning C4244: '=': conversion from '_Ty' to 'uint8_t', possible loss of data
          with
          [
              _Ty=float
          ]
  ggwave/src/ggwave.cpp(663): warning C4244: '=': conversion from '_Ty' to 'uint16_t', possible loss of data
          with
          [
              _Ty=float
          ]
  ggwave/src/ggwave.cpp(1003): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
  ggwave/src/ggwave.cpp(1106): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
  ggwave/src/ggwave.cpp(1155): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
  ggwave/src/ggwave.cpp(1133): warning C4996: 'asctime': This function or variable may be unsafe. Consider using asctime_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  ggwave/src/ggwave.cpp(1182): warning C4996: 'asctime': This function or variable may be unsafe. Consider using asctime_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  ggwave/src/ggwave.cpp(1224): warning C4267: '+=': conversion from 'size_t' to 'int', possible loss of data
  ggwave/src/ggwave.cpp(1247): warning C4267: '-=': conversion from 'size_t' to 'int', possible loss of data
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30037\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
  ----------------------------------------
  ERROR: Failed building wheel for ggwave
  Running setup.py clean for ggwave
Failed to build ggwave
Installing collected packages: ggwave
    Running setup.py install for ggwave ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\meetr\appdata\local\programs\python\python38\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-install-azsi740u\\ggwave_5580c82224ed49db82087eca3795e92a\\setup.py'"'"'; __file__='"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-install-azsi740u\\ggwave_5580c82224ed49db82087eca3795e92a\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Meetr\AppData\Local\Temp\pip-record-hja6gpmf\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\meetr\appdata\local\programs\python\python38\Include\ggwave'
         cwd: C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\
    Complete output (73 lines):
    running install
    running build
    running build_ext
    building 'ggwave' extension
    creating build
    creating build\temp.win-amd64-3.8
    creating build\temp.win-amd64-3.8\Release
    creating build\temp.win-amd64-3.8\Release\ggwave
    creating build\temp.win-amd64-3.8\Release\ggwave\src
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Iggwave/include -Iggwave/include/ggwave -Ic:\users\meetr\appdata\local\programs\python\python38\include -Ic:\users\meetr\appdata\local\programs\python\python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt" /EHsc /Tpggwave.bycython.cpp /Fobuild\temp.win-amd64-3.8\Release\ggwave.bycython.obj -O3 -std=c++11
    cl : Command line warning D9002 : ignoring unknown option '-O3'
    cl : Command line warning D9002 : ignoring unknown option '-std=c++11'
    ggwave.bycython.cpp
    ggwave.bycython.cpp(1881): warning C4244: 'argument': conversion from 'Py_ssize_t' to 'int', possible loss of data
    ggwave.bycython.cpp(1923): warning C4244: 'argument': conversion from 'Py_ssize_t' to 'int', possible loss of data
    ggwave.bycython.cpp(2162): warning C4244: 'argument': conversion from 'Py_ssize_t' to 'int', possible loss of data
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Iggwave/include -Iggwave/include/ggwave -Ic:\users\meetr\appdata\local\programs\python\python38\include -Ic:\users\meetr\appdata\local\programs\python\python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt" /EHsc /Tpggwave/src/ggwave.cpp /Fobuild\temp.win-amd64-3.8\Release\ggwave/src/ggwave.obj -O3 -std=c++11
    cl : Command line warning D9002 : ignoring unknown option '-O3'
    cl : Command line warning D9002 : ignoring unknown option '-std=c++11'
    ggwave.cpp
    C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon\gf.hpp(150): warning C4244: 'argument': conversion from 'uint16_t' to 'uint8_t', possible loss of data
    C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon\gf.hpp(150): warning C4244: 'argument': conversion from 'uint16_t' to 'uint8_t', possible loss of data
    C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon\gf.hpp(216): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data
    C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon/rs.hpp(174): warning C4267: 'argument': conversion from 'size_t' to 'uint8_t', possible loss of data
    C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon/rs.hpp(455): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data
    C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon/rs.hpp(500): warning C4267: 'argument': conversion from 'size_t' to 'uint8_t', possible loss of data
    C:\Users\Meetr\AppData\Local\Temp\pip-install-azsi740u\ggwave_5580c82224ed49db82087eca3795e92a\ggwave\src\reed-solomon/rs.hpp(519): warning C4267: 'argument': conversion from 'size_t' to 'uint8_t', possible loss of data
    ggwave/src/ggwave.cpp(172): error C2065: 'M_PI': undeclared identifier
    ggwave/src/ggwave.cpp(173): error C2065: 'M_PI': undeclared identifier
    ggwave/src/ggwave.cpp(177): error C2065: 'M_PI': undeclared identifier
    ggwave/src/ggwave.cpp(178): error C2065: 'M_PI': undeclared identifier
    ggwave/src/ggwave.cpp(229): warning C4244: 'initializing': conversion from 'float' to 'int', possible loss of data
    ggwave/src/ggwave.cpp(230): warning C4244: 'initializing': conversion from 'float' to 'int', possible loss of data
    ggwave/src/ggwave.cpp(232): warning C4244: 'initializing': conversion from 'int' to 'float', possible loss of data
    ggwave/src/ggwave.cpp(337): warning C4305: 'initializing': truncation from 'double' to 'float'
    ggwave/src/ggwave.cpp(385): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
    ggwave/src/ggwave.cpp(389): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
    ggwave/src/ggwave.cpp(415): warning C4244: '=': conversion from 'double' to 'float', possible loss of data
    ggwave/src/ggwave.cpp(499): error C2065: 'M_PI': undeclared identifier
    ggwave/src/ggwave.cpp(524): error C2065: 'M_PI': undeclared identifier
    ggwave/src/ggwave.cpp(528): error C2065: 'M_PI': undeclared identifier
    ggwave/src/ggwave.cpp(639): warning C4244: '=': conversion from '_Ty' to '_Ty', possible loss of data
            with
            [
                _Ty=float
            ]
            and
            [
                _Ty=int16_t
            ]
    ggwave/src/ggwave.cpp(649): warning C4244: '=': conversion from '_Ty' to 'uint8_t', possible loss of data
            with
            [
                _Ty=float
            ]
    ggwave/src/ggwave.cpp(656): warning C4244: '=': conversion from '_Ty' to 'uint8_t', possible loss of data
            with
            [
                _Ty=float
            ]
    ggwave/src/ggwave.cpp(663): warning C4244: '=': conversion from '_Ty' to 'uint16_t', possible loss of data
            with
            [
                _Ty=float
            ]
    ggwave/src/ggwave.cpp(1003): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
    ggwave/src/ggwave.cpp(1106): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
    ggwave/src/ggwave.cpp(1155): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
    ggwave/src/ggwave.cpp(1133): warning C4996: 'asctime': This function or variable may be unsafe. Consider using asctime_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
    ggwave/src/ggwave.cpp(1182): warning C4996: 'asctime': This function or variable may be unsafe. Consider using asctime_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
    ggwave/src/ggwave.cpp(1224): warning C4267: '+=': conversion from 'size_t' to 'int', possible loss of data
    ggwave/src/ggwave.cpp(1247): warning C4267: '-=': conversion from 'size_t' to 'int', possible loss of data
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30037\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\meetr\appdata\local\programs\python\python38\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-install-azsi740u\\ggwave_5580c82224ed49db82087eca3795e92a\\setup.py'"'"'; __file__='"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-install-azsi740u\\ggwave_5580c82224ed49db82087eca3795e92a\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Meetr\AppData\Local\Temp\pip-record-hja6gpmf\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\meetr\appdata\local\programs\python\python38\Include\ggwave' Check the logs for full command output.

**Comment by ggerganov** [comment]
Hi, can you try adding the following snippet to [src/ggwave.cpp](https://github.com/ggerganov/ggwave/blob/59f576f1e8132bddc865524209dd37a7fa4e221c/src/ggwave.cpp#L13) after the includes at the beginning of the file:

```c++
#ifndef M_PI
    #define M_PI 3.14159265358979323846
#endif
```

I don't have a Windows setup and cannot try it at the moment.

**Comment by meetrajsinh312** [comment]
> Hi, can you try adding the following snippet to [src/ggwave.cpp](https://github.com/ggerganov/ggwave/blob/59f576f1e8132bddc865524209dd37a7fa4e221c/src/ggwave.cpp#L13) after the includes at the beginning of the file:
> 
> ```c++
> #ifndef M_PI
>     #define M_PI 3.14159265358979323846
> #endif
> ```
> 
> I don't have a Windows setup and cannot try it at the moment.

I tried this with all versions of ggwave still not fixed:::

c1xx: fatal error C1083: Cannot open source file: 'ggwave.bycython.cpp': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30037\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2

full error:::
C:\WINDOWS\system32>pip install D:\ggwave-master\bindings\python
Processing d:\ggwave-master\bindings\python
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
Building wheels for collected packages: ggwave
  Building wheel for ggwave (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\users\meetr\appdata\local\programs\python\python38\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-req-build-n7_7xy_a\\setup.py'"'"'; __file__='"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-req-build-n7_7xy_a\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\Meetr\AppData\Local\Temp\pip-wheel-k4s3at54'
       cwd: C:\Users\Meetr\AppData\Local\Temp\pip-req-build-n7_7xy_a\
  Complete output (15 lines):
  running bdist_wheel
  running build
  running build_ext
  building 'ggwave' extension
  creating build
  creating build\temp.win-amd64-3.8
  creating build\temp.win-amd64-3.8\Release
  creating build\temp.win-amd64-3.8\Release\ggwave
  creating build\temp.win-amd64-3.8\Release\ggwave\src
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Iggwave/include -Iggwave/include/ggwave -Ic:\users\meetr\appdata\local\programs\python\python38\include -Ic:\users\meetr\appdata\local\programs\python\python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt" /EHsc /Tpggwave.bycython.cpp /Fobuild\temp.win-amd64-3.8\Release\ggwave.bycython.obj -O3 -std=c++11
  cl : Command line warning D9002 : ignoring unknown option '-O3'
  cl : Command line warning D9002 : ignoring unknown option '-std=c++11'
  ggwave.bycython.cpp
  c1xx: fatal error C1083: Cannot open source file: 'ggwave.bycython.cpp': No such file or directory
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30037\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
  ----------------------------------------
  ERROR: Failed building wheel for ggwave
  Running setup.py clean for ggwave
Failed to build ggwave
Installing collected packages: ggwave
    Running setup.py install for ggwave ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\meetr\appdata\local\programs\python\python38\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-req-build-n7_7xy_a\\setup.py'"'"'; __file__='"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-req-build-n7_7xy_a\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Meetr\AppData\Local\Temp\pip-record-t7bxpw2a\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\meetr\appdata\local\programs\python\python38\Include\ggwave'
         cwd: C:\Users\Meetr\AppData\Local\Temp\pip-req-build-n7_7xy_a\
    Complete output (15 lines):
    running install
    running build
    running build_ext
    building 'ggwave' extension
    creating build
    creating build\temp.win-amd64-3.8
    creating build\temp.win-amd64-3.8\Release
    creating build\temp.win-amd64-3.8\Release\ggwave
    creating build\temp.win-amd64-3.8\Release\ggwave\src
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Iggwave/include -Iggwave/include/ggwave -Ic:\users\meetr\appdata\local\programs\python\python38\include -Ic:\users\meetr\appdata\local\programs\python\python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt" /EHsc /Tpggwave.bycython.cpp /Fobuild\temp.win-amd64-3.8\Release\ggwave.bycython.obj -O3 -std=c++11
    cl : Command line warning D9002 : ignoring unknown option '-O3'
    cl : Command line warning D9002 : ignoring unknown option '-std=c++11'
    ggwave.bycython.cpp
    c1xx: fatal error C1083: Cannot open source file: 'ggwave.bycython.cpp': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30037\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\meetr\appdata\local\programs\python\python38\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-req-build-n7_7xy_a\\setup.py'"'"'; __file__='"'"'C:\\Users\\Meetr\\AppData\\Local\\Temp\\pip-req-build-n7_7xy_a\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Meetr\AppData\Local\Temp\pip-record-t7bxpw2a\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\meetr\appdata\local\programs\python\python38\Include\ggwave' Check the logs for full command output.

**Comment by ggerganov** [comment]
I just released new ggwave version `0.3.0`.
Please, try running `pip install ggwave` again and see if the problem is fixed.

**Comment by meetrajsinh312** [comment]
> I just released new ggwave version `0.3.0`.
> Please, try running `pip install ggwave` again and see if the problem is fixed.

Issue fixed 
thank you so much for quick response i was trying this since last 4 days.

**Comment by ggerganov** [comment]
Thanks for reporting this - I rarely use python, so there might be some more problems with ggwave that I am not aware of.
Feel free to open new issues if you encounter any problems!
